### PR TITLE
Code standardization and dOxyGenization of framework code

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
Extensive dOxyGenization and cleanup of MOM6 code, including the following commits:

 - NOAA-GFDL/MOM6@6e018cf +Combined arguments to write_[uv]_accel
 - NOAA-GFDL/MOM6@4bdfed6 Shortened excesively long lines in ice_solo_driver
 - NOAA-GFDL/MOM6@881bf4c dOxyGenized Neverland_surface_forcing.F90
 - NOAA-GFDL/MOM6@01b434e dOxyGenized MESO_surface_forcing.F90
 - NOAA-GFDL/MOM6@7ff278e dOxyGenized user_surface_forcing.F90
 - NOAA-GFDL/MOM6@5663737 dOxyGenized solo_driver/MOM_surface_forcing.F90
 - NOAA-GFDL/MOM6@a4990de dOxyGenized MOM_sum_output.F90
 - NOAA-GFDL/MOM6@28da3a2 dOxyGenized MOM_grid_initialization.F90
 - NOAA-GFDL/MOM6@1d10503 dOxyGenized MOM_wave_structure.F90
 - NOAA-GFDL/MOM6@f5fa06c dOxyGenized MOM_obsolete_diagnostics.F90
 - NOAA-GFDL/MOM6@3c95708 dOxyGenized MOM_diagnostics.F90
 - NOAA-GFDL/MOM6@462595f Removed trailing white space
 - NOAA-GFDL/MOM6@536382e dOxyGenized MOM_write_cputime.F90
 - NOAA-GFDL/MOM6@434f77c dOxyGenized MOM_string_functions.F90
 - NOAA-GFDL/MOM6@397d258 dOxyGenized MOM_restart.F90
 - NOAA-GFDL/MOM6@d9e8f1f dOxyGenized MOM_io.F90
 - NOAA-GFDL/MOM6@396b9bc dOxyGenized MOM_intrinsic_functions.F90
 - NOAA-GFDL/MOM6@6c198b7 dOxyGenized MOM_domains.F90
 - NOAA-GFDL/MOM6@dc91dd0 dOxyGenized MOM_diag_remap.F90
 - NOAA-GFDL/MOM6@70f96c1 dOxyGenized MOM_diag_mediator.F90
 - NOAA-GFDL/MOM6@3bc5d09 Split excessively long lines in MOM_file_parser
 - NOAA-GFDL/MOM6@668f13a Split excessively long lines in MOM_open_boundary
 - NOAA-GFDL/MOM6@a9f84c4 Fixed some dOxygen comments in MOM_debugging.F90
 - NOAA-GFDL/MOM6@e8e2459 Merge branch 'dev/gfdl' into code_standardization
 - NOAA-GFDL/MOM6@ce614c6 dOxyGenized MOM_spatial_means.F90
 - NOAA-GFDL/MOM6@5490e5a dOxyGenized MOM_coms.F90
 - NOAA-GFDL/MOM6@0116078 Merged keywords end do & end if to enddo & endif

### CODE REVIEW

**Comment by adcroft** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/4130

File: src/framework/MOM_diag_mediator.F90 [hunk 1]
```diff
@@ -2510,24 +2547,26 @@ subroutine diag_mediator_end(time, diag_CS, end_diag_manager)
...
+  character(len=15) :: i2s !< The returned string
+
+  character(len=15) :: i2s_temp
+  integer :: i,n
+
+  n=size(a)
+  if (present(n_in)) n = n_in
+
+  i2s = ''
+  do i=1,max(n,3)
```
Comment (angus-g) [inline src/framework/MOM_diag_mediator.F90]: `do i=1,min(n,3)` ?

**Comment by Hallberg-NOAA** [comment]
5 additional commits have corrected a bug (thanks @angus-g), merged in a recent commit to dev/gfdl, and added updates dOxyGenizing much of the MOM6 tracer code.

**Comment by adcroft** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/4135

**Pull request merged**